drivers/gpio/gpio_mchp_xec.c

```diff
@@ -70,83 +83,150 @@ static int gpio_xec_configure(struct device *dev,
 	mask |= MCHP_GPIO_CTRL_DIR_MASK;
 	mask |= MCHP_GPIO_CTRL_INPAD_DIS_MASK;
 	if (access_op == GPIO_ACCESS_BY_PIN) {
-		if ((flags & GPIO_DIR_MASK) == GPIO_DIR_IN) {
-			pcr1 &= ~BIT(MCHP_GPIO_CTRL_DIR_POS);
-		} else { /* GPIO_DIR_OUT */
-			pcr1 |= BIT(MCHP_GPIO_CTRL_DIR_POS);
+		if ((flags & GPIO_OUTPUT) != 0U) {
+			if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0U) {
+				*gpio_base |= BIT(pin);
+			} else if ((flags & GPIO_OUTPUT_INIT_LOW) != 0U) {
+				*gpio_base &= ~BIT(pin);
+			}
+			pcr1 |= MCHP_GPIO_CTRL_DIR_OUTPUT;
+		} else {
+			/* GPIO_INPUT */
+			pcr1 |= MCHP_GPIO_CTRL_DIR_INPUT;
 		}
 	} else { /* GPIO_ACCESS_BY_PORT is not supported */
-		return -EINVAL;
+		return -ENOTSUP;
 	}
 
 	/* Figure out the pullup/pulldown configuration and keep it in the
 	 * pcr1 variable
 	 */
 	mask |= MCHP_GPIO_CTRL_PUD_MASK;
 
-	if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_UP) {
+	if ((flags & GPIO_PULL_UP) != 0U) {
 		/* Enable the pull and select the pullup resistor. */
 		pcr1 |= MCHP_GPIO_CTRL_PUD_PU;
-	} else if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_DOWN) {
+	} else if ((flags & GPIO_PULL_DOWN) != 0U) {
 		/* Enable the pull and select the pulldown resistor */
 		pcr1 |= MCHP_GPIO_CTRL_PUD_PD;
 	}
 
+	/* Push-pull or open drain */
+
+	if ((flags & GPIO_OPEN_DRAIN) != 0U) {
+		/* Open drain */
+		pcr1 |= MCHP_GPIO_CTRL_BUFT_OPENDRAIN;
+		mask |= MCHP_GPIO_CTRL_BUFT_MASK;
+	} else {
+		/* Push-pull */
+		pcr1 |= MCHP_GPIO_CTRL_BUFT_PUSHPULL;
+		mask |= MCHP_GPIO_CTRL_BUFT_MASK;
+	}
+
+	/* Use Gpio output register for writing in order to have simetry with
+	 * respect of Gpio input
+	 */
+	mask |= MCHP_GPIO_CTRL_AOD_MASK;
+	pcr1 |= MCHP_GPIO_CTRL_AOD_MASK;
```

Comment:
This is an old code, however it would be better to write it as
```
	pcr1 |= MCHP_GPIO_CTRL_AOD_DIS;
```
And I would also change the comment. Maybe to something like:
"Disable alternate output, control pin output using the parallel GPIO output register"